sqlglot/dialects/tsql.py

```diff
@@ -205,6 +234,7 @@ class Generator(Generator):
 
         TRANSFORMS = {
             **Generator.TRANSFORMS,
-            exp.DateAdd: lambda self, e: f"DATEADD({self.format_args(e.args.get('unit'), e.expression, e.this)})",
+            exp.DateAdd: lambda self, e: generate_date_delta(self, e),
+            exp.DateDiff: lambda self, e: f"DATEDIFF({self.format_args(e.args.get('unit'), e.expression, e.this)})",
```

Comment:
We should be able to reuse generate_date_delta for DateDiff too, right? Also, maybe rename generate_date_delta --> _date_delta_sql for consistency.
```python
exp.DateAdd: generate_date_delta,
```

Reply:
Yes you're right